test/test_gcs_obj_metadata_client.py

```diff
@@ -16,35 +16,135 @@ def run(self):
 
 
 class TestGCSObjectMetadataClient(unittest.TestCase):
-    def test_get_patched_obj_metadata(self):
-        task_params: dict[Any, str] = {
+    def setUp(self):
+        self.task_params: dict[Any, str] = {
             'param1': 'a' * 1000,
             'param2': str(1000),
             'param3': str({'key1': 'value1', 'key2': True, 'key3': 2}),
             'param4': str([1, 2, 3, 4, 5]),
             'param5': str(datetime.datetime(year=2025, month=1, day=2, hour=3, minute=4, second=5)),
             'param6': '',
         }
-        got = GCSObjectMetadataClient._get_patched_obj_metadata({}, task_params=task_params)
+        self.custom_labels: dict[Any, Any] = {
+            'created_at': datetime.datetime(year=2025, month=1, day=2, hour=3, minute=4, second=5),
+            'created_by': 'hoge fuga',
+            'empty': True,
+            'try_num': 3,
+        }
+
+        self.task_params_with_conflicts = {
+            'empty': 'False',
+            'created_by': 'fuga hoge',
+            'param1': 'a' * 10,
+        }
+
+    def test_normalize_labels_both_are_empty(self):
+        got_norm_task_params_labels = GCSObjectMetadataClient._normalize_labels(None)
+        got_norm_custom_labels = GCSObjectMetadataClient._normalize_labels(None)
+        self.assertIsInstance(got_norm_task_params_labels, dict)
+        self.assertIsInstance(got_norm_custom_labels, dict)
+        self.assertEqual(got_norm_task_params_labels, {})
+        self.assertEqual(got_norm_custom_labels, {})
+
+    def test_normalize_labels_only_task_params(self):
+        got_norm_task_params_labels = GCSObjectMetadataClient._normalize_labels(self.task_params)
+        got_norm_custom_labels = GCSObjectMetadataClient._normalize_labels(None)
```

Comment:
[should]
we do not need to tests combinations `task_params` and `custom_labels` on this function,
because `_normalize_labels` does not aware them.

please focus on how `_normalize_labels` converts optional[dict[str, Any]] to dict[str, str].